flink-core/src/main/java/org/apache/flink/util/concurrent/DirectExecutorService.java

```diff
@@ -39,10 +41,37 @@
  * thread.
  */
 class DirectExecutorService implements ExecutorService {
-    static final DirectExecutorService INSTANCE = new DirectExecutorService();
+
+    private final ThrowingRunnable<? extends RuntimeException> ifShutdownAction;
```

Comment:
Wouldn't a boolean flag do the trick as well + make this easier to interpret? :)